src/neuroconv/tools/nwb_helpers/_metadata_and_file_helpers.py

```diff
@@ -221,27 +222,35 @@ def make_or_load_nwbfile(
 
         io = backend_io_class(**load_kwargs)
 
+    nwbfile_loaded_succesfully = True
     try:
         if load_kwargs.get("mode", "") == "r+":
             nwbfile = io.read()
         elif nwbfile is None:
             nwbfile = make_nwbfile_from_metadata(metadata=metadata)
         yield nwbfile
-    except Exception as e:
-        success = False
-        raise e
+    except Exception as load_error:
+        nwbfile_loaded_succesfully = False
+        raise load_error
     finally:
         if nwbfile_path_in:
+            nwbfile_written_succesfully = True
             try:
-                if success:
+                if nwbfile_loaded_succesfully:
                     io.write(nwbfile)
 
                     if verbose:
                         print(f"NWB file saved at {nwbfile_path_in}!")
+            except Exception as write_error:
+                nwbfile_written_succesfully = False
+                raise write_error
```

Comment:
I see, you also added a capture block here to further scope the success of the operation

Reply:
Right. I think this is the core. Separating errors from loading or creating from errors of writing.

But as you were mentioning maybe us handling unlinking is not a good idea ...

Reply:
In this case we're attempting to leave the system as we left it, which is fine

The case I am against is what you don't have here

```python
if any_error_condition and file_initially_exists:
    nwbfile_path_in.unlink()
```